### PR TITLE
1 implement hex grid function

### LINKED ISSUE #1: Implement hex grid function
First port of call: h3o

Heads-up @JosiahParry here's a use case for you, and maybe a motivator for getting this published on CRAN which I'm happy to help with.

Need to check it actually works for our use case first, hope so!

### PR DESCRIPTION
**Pull request by Robinlovelace**
- Add tests for #1
- Add h3o, close #1

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Comments welcome @josiahparry

File: R/iso_grid.R [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
-#' @param ... Extra arguments passed to the grid generator (TBC)
+#' @param resolution The resolution of the hex grid
+#' @return An sf object
 #' @export
 #' @examples
 #' x = net_oldenburg_raw
 #' iso_grid(x)
-iso_grid = function(x, ...) {
-  sf::st_convex_hull(sf::st_union(x))
+iso_grid = function(x, resolution = 9) {
```
Comment (bda1986) [inline R/iso_grid.R:12]: is resolution = 9 chosen arbitrarily or does it reflect some pre-defined size?

File: R/iso_grid.R [hunk 2]
```diff
@@ -1,11 +1,19 @@
 #' Create hexagonal grid
+#' 
+#' Create a hexagonal grid from an sf object
 #'
 #' @param x An sf object the convex hull of which will determine the hex grid
-#' @param ... Extra arguments passed to the grid generator (TBC)
+#' @param resolution The resolution of the hex grid
+#' @return An sf object
```
Comment (JosiahParry) [inline R/iso_grid.R:7]: This returns an `sfc` class object not an `sf` object

Comment (Robinlovelace) [reply R/iso_grid.R:7]: Ah yes. Fix incoming.

File: R/iso_grid.R [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
 #' @export
 #' @examples
 #' x = net_oldenburg_raw
 #' iso_grid(x)
-iso_grid = function(x, ...) {
-  sf::st_convex_hull(sf::st_union(x))
+iso_grid = function(x, resolution = 9) {
+  poly = sf::st_convex_hull(sf::st_union(x))
```
Comment (JosiahParry) [inline R/iso_grid.R:13]: Probably best for me to make the check in h3o, but there needs to be validation that the units are degrees

Comment (Robinlovelace) [reply R/iso_grid.R:13]: Yeah... safe to assume EPSG:4326 I guess. We can mention that in the docs, if that would fix it?

Comment (Robinlovelace) [reply R/iso_grid.R:13]: And at some point could shift to the whole `{rsgeo}` stack, but let's get something working first!

**Review by bda1986 — APPROVED** [review]

**Comment by Robinlovelace** [comment]
Going to merge

**Pull request merged**